src/main/java/org/apache/pirk/querier/wideskies/QuerierProps.java

```diff
@@ -0,0 +1,202 @@
+package org.apache.pirk.querier.wideskies;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.pirk.schema.data.LoadDataSchemas;
+import org.apache.pirk.schema.query.LoadQuerySchemas;
+import org.apache.pirk.utils.SystemConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Properties constants and validation for the Querier
+ */
+public class QuerierProps
+{
+  private static final Logger logger = LoggerFactory.getLogger(QuerierProps.class);
+
+  // General properties
+  public static final String ACTION = "querier.action";
+  public static final String INPUTFILE = "querier.inputFile";
+  public static final String OUTPUTFILE = "querier.outputFile";
+  public static final String QUERYTYPE = "querier.queryType";
+  public static final String NUMTHREADS = "querier.numThreads";
+
+  // Encryption properties
+  public static final String HASHBITSIZE = "querier.hashBitSize";
+  public static final String HASHKEY = "querier.hashKey";
+  public static final String DATAPARTITIONSIZE = "querier.dataPartitionBitSize";
+  public static final String PAILLIERBITSIZE = "querier.paillierBitSize";
+  public static final String BITSET = "querier.bitSet";
+  public static final String CERTAINTY = "querier.certainty";
+  public static final String QUERYNAME = "querier.queryName";
+  public static final String QUERYSCHEMAS = "querier.querySchemas";
+  public static final String DATASCHEMAS = "querier.dataSchemas";
+  public static final String EMBEDSELECTOR = "querier.embedSelector";
+  public static final String USEMEMLOOKUPTABLE = "querier.memLookupTable";
+  public static final String USEHDFSLOOKUPTABLE = "querier.useHDFSLookupTable";
+  public static final String SR_ALGORITHM = "pallier.secureRandom.algorithm";
+  public static final String SR_PROVIDER = "pallier.secureRandom.provider";
+  public static final String EMBEDQUERYSCHEMA = "pir.embedQuerySchema";
+
+  // Decryption properties
+  public static final String QUERIERFILE = "querier.querierFile";
+
+  public static final List<String> PROPSLIST = Arrays.asList(ACTION, INPUTFILE, OUTPUTFILE, QUERYTYPE, NUMTHREADS, EMBEDQUERYSCHEMA, HASHBITSIZE, HASHKEY,
+      DATAPARTITIONSIZE, PAILLIERBITSIZE, BITSET, CERTAINTY, QUERYNAME, QUERYSCHEMAS, DATASCHEMAS, EMBEDSELECTOR, USEMEMLOOKUPTABLE, USEHDFSLOOKUPTABLE,
+      SR_ALGORITHM, SR_PROVIDER);
+
+  /**
+   * Validates the querier properties
+   * 
+   */
+  public static boolean validateQuerierProperties()
```

Comment:
How about checking the required properties in bulk? e.g. something like...

```
        static final String[] REQUIRED_PROPERTIES = new String[] {QUERYTYPE, HASHBITSIZE, ETC}; 
        ....
        List<String> missingProperties = Arrays.asList(REQUIRED_PROPERTIES).removeAll(SystemConfiguration.getPropertyNames());
        boolean valid = missingProperties.isEmpty(); 
        for (String property : missingProperties) {
          logger.info("Must have the option " + property);
        }
```
(You will have to persuade ```SystemConfiguration``` to tell you all the property names.)